File: C#/Refactoring/refactoring_exercise_1/za/co/entelect/refactoring1/controller/ImageServiceDelegate.cs
using System;
using refactoring_exercise_1.za.co.entelect.refactoring1.domain;
using refactoring_exercise_1.za.co.entelect.refactoring1.service;

namespace refactoring_exercise_1.za.co.entelect.refactoring1.controller
{

    public class ImageServiceDelegate
    {

        private ImageService imageService = new ImageService();

        public Image Fetch(String id)
        {
            return imageService.Fetch(id);
        }

        public int Count()
        {
            return imageService.Count();
        }

        public void Add(Image image)
        {
            imageService.Add(image);
        }
    }
}

File: Java/refactoring_exercise_1/src/main/java/za/co/entelect/refactoring1/controller/ImageServiceDelegate.java
package za.co.entelect.refactoring1.controller;

import za.co.entelect.refactoring1.domain.Image;
import za.co.entelect.refactoring1.service.ImageService;


public class ImageServiceDelegate {

   private ImageService imageService = new ImageService();

    public Image fetch(String id) {
        return imageService.fetch(id);
    }

    public int count() {
        return imageService.count();
    }

    public void add(Image image) {
        imageService.add(image);
    }
}
